### PR TITLE
chore: upgrade threejs to 0.125.0

### PR DESCRIPTION
**Pull request by psybuzz**
Upgrading to the newer Three.js 0.125.0 to silence dependabot warnings.
Notable changes:
- `Material`'s `vertexColors` is now a boolean
- `BufferAttribute.setArray` was removed in favor of
   `BufferedGeometry.setAttribute`
- Words like `texture` became reserved keywords when used in Three.js'
   shaders

Manually checked that the mesh demo, projector demo, and some custom
projector example with sprite images look as expected. I did not test the
"polylines that connect multiple points in a dataset" feature in the Projector
since I do not know how it works. Line charts in the Time Series look normal
as well.

See the Three.js changelog: https://github.com/mrdoob/three.js/wiki/Migration-Guide
Will be rendered obsolete: https://github.com/tensorflow/tensorboard/pull/4715
Googlers, see b/150083780, test sync cl/364427588, and copybara
cl/365095009.

image

### CODE REVIEW

File: tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts [hunk 1]
```diff
@@ -441,46 +435,37 @@ export class MeshViewer extends THREE.EventDispatcher {
...
-            vertex_color[2] / 255
-          );
-          face.vertexColors.push(color);
-        }
-      }
-      geometry.faces.push(face);
-    });
+    const geometry = new THREE.BufferGeometry();
+    // Flattens the [N, 3] shaped `vertices` into a N*3 length array.
+    const flatPoints = new Float32Array(vertices.flat());
```
Comment (japie1235813) [inline tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts:440]: Could you add more explanation why we flattens here?
Not sure this is equivalent to L445-452.

File: tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts [hunk 2]
```diff
@@ -441,46 +435,37 @@ export class MeshViewer extends THREE.EventDispatcher {
...
+    geometry.setAttribute('position', new THREE.BufferAttribute(flatPoints, 3));
+
+    // Flattens the [N, 3] shaped `faces` into a N*3 length array of indices.
+    const vertexIndicesPerFace = new Uint16Array(faces.flat());
```
Comment (japie1235813) [inline tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts:444]: Why "face" is an integer? (while points are float) What is the meaning of it?

**Review by japie1235813 — APPROVED** [review]

File: tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts (hunk 1 above)
Comment (psybuzz) [reply tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts:442]: Updated the comment to explain that THREE.BufferAttribute expects a flat array.

File: tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts (hunk 2 above)
Comment (psybuzz) [reply tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-viewer.ts:457]: Updated the comment.

This array contains indices which correspond to "items" in the 'positions' attribute.  Added an example in the comment to illustrate how each "item" is a triplet of values (XYZ).

Most of my understanding was gained through reading this helpful site, and debugging
https://threejsfundamentals.org/threejs/lessons/threejs-custom-buffergeometry.html

**Review by psybuzz — COMMENTED** [review]
Thanks for reviewing!

**Pull request merged**